vault/resource_ldap_secret_backend.go

```diff
@@ -0,0 +1,684 @@
+		}
+	}
+	if val, ok := resp.Data["tls_max_version"]; ok {
+		if err := d.Set("tls_max_version", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'tls_max_version': %s", err))
+		}
+	}
+	if val, ok := resp.Data["tls_min_version"]; ok {
+		if err := d.Set("tls_min_version", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'tls_min_version': %s", err))
+		}
+	}
+	if val, ok := resp.Data["ttl"]; ok {
+		if err := d.Set("ttl", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'ttl': %s", err))
+		}
+	}
+	if val, ok := resp.Data["upndomain"]; ok {
+		if err := d.Set("upndomain", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'upndomain': %s", err))
+		}
+	}
+	if val, ok := resp.Data["url"]; ok {
+		if err := d.Set("url", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'url': %s", err))
+		}
+	}
+	if val, ok := resp.Data["use_pre111_group_cn_behavior"]; ok {
+		if err := d.Set("use_pre111_group_cn_behavior", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'use_pre111_group_cn_behavior': %s", err))
+		}
+	}
+	if val, ok := resp.Data["use_token_groups"]; ok {
+		if err := d.Set("use_token_groups", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'use_token_groups': %s", err))
+		}
+	}
+	if val, ok := resp.Data["userattr"]; ok {
+		if err := d.Set("userattr", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'userattr': %s", err))
+		}
+	}
+	if val, ok := resp.Data["userdn"]; ok {
+		if err := d.Set("userdn", val); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'userdn': %s", err))
+		}
+	}
+	return nil
+}
+
+func updateLDAPConfigResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	backend := d.Id()
+
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	backend, err = util.Remount(d, client, consts.FieldBackend, false)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	defaultTTL := d.Get("default_lease_ttl_seconds").(int)
+	maxTTL := d.Get("max_lease_ttl_seconds").(int)
+	tune := api.MountConfigInput{}
+	data := map[string]interface{}{}
+
+	if defaultTTL != 0 {
+		tune.DefaultLeaseTTL = fmt.Sprintf("%ds", defaultTTL)
+		data["default_lease_ttl_seconds"] = defaultTTL
+	}
+
+	if maxTTL != 0 {
+		tune.MaxLeaseTTL = fmt.Sprintf("%ds", maxTTL)
+		data["max_lease_ttl_seconds"] = maxTTL
+	}
+
+	if tune.DefaultLeaseTTL != "0" || tune.MaxLeaseTTL != "0" {
+		err := client.Sys().TuneMount(backend, tune)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error mounting to %q: %s", backend, err))
+		}
+	}
+
+	vaultPath := fmt.Sprintf("%s/config", backend)
+	log.Printf("[DEBUG] Updating %q", vaultPath)
+
+	if raw, ok := d.GetOk("anonymous_group_search"); ok {
```

Comment:
We can probably consolidate the create and update methods into a single method, and use the `d.IsNewResource()` util to determine whether we are in a create or update call. An example of how the consolidation can be done is [provided here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_kubernetes_secret_backend.go#L79-L89)